packages/instantsearch.js/src/__tests__/common-widgets.test.tsx

```diff
@@ -418,7 +418,7 @@ const testSetups: TestSetupsMap<TestSuites> = {
           container: document.body.appendChild(document.createElement('div')),
           ...widgetParams,
           templates: {
-            labelText: label,
+            labelText: label ? () => label : undefined,
```

Comment:
does this imply that string templates aren't supported anymore? could we check if it's a string and just render as a string instead of requiring a function?

Reply:
That's what I thought of doing at first until I saw we print a warning when string templates are used, saying these will be deprecated : https://github.com/algolia/instantsearch/blob/b24c858b66de50d8dde6c85ce7ffc90ab0f712e8/packages/instantsearch.js/src/components/Template/Template.tsx#L77C1-L91C6

Another upside of keeping function templates only is they would easily see it break if they keep hogan or a HTML string when upgrading, whereas if we still support strings it could easily be missed (`<p>Something</p>` rendered as escaped HTML for example)

Reply:
The warning was mainly to deprecate hogan strings as we couldn't tell them apart from simple strings. I see how it's simpler for us, but the migration for simple strings (most of the files changed in this pr) shows that it might be annoying for customers to do. What do you think?